scripts/uosc/elements/Menu.lua

```diff
@@ -131,12 +131,14 @@ function Menu:destroy()
 	self:disable_key_bindings()
 	self.is_closed = true
 	if not self.is_being_replaced then Elements.curtain:unregister('menu') end
+	mp.del_property('user-data/uosc/menu/type')
 	if self.opts.on_close then self.opts.on_close() end
 end
 
 ---@param data MenuData
 function Menu:update(data)
 	self.type = data.type
+	mp.set_property_native('user-data/uosc/menu/type', data.type)
```

Comment:
Is `set_property_native` going to trigger observers if the property is assigned the same value? If so, I don't think this should be in `:update()`. It's not like menu type can change. And if it can somehow, it's a bug.

Reply:
I think it triggers observers, didn't test. But even if it doesn't, `init()` is a better place for it.